UnityProject/Assets/Scripts/Objects/Canister.cs

```diff
@@ -28,7 +30,7 @@ public override bool Interact(GameObject originator, Vector3 position, string ha
 		{
 			if(connector == null)
 			{
-				var foundConnectors = MatrixManager.GetAt<Connector>(position.RoundToInt(), true);
+				var foundConnectors = MatrixManager.GetAt<Connector>(registerTile.PositionServer, true);
```

Comment:
MatrixManager works with World positions only – it iterates through all matrices (And `Matrix` class is for local positions), use `WorldPositionServer` here